### PR TITLE
codebase/using-amazon-textract-in-spring-boot-to-extract-text-from-images [BAEL-6759]

### PR DESCRIPTION
**Pull request by hardikSinghBehl**
Codebase for the article: **Using Amazon Textract in Spring Boot to Extract Text From Images**

### CODE REVIEW

File: aws-modules/amazon-textract/src/main/java/com/baeldung/textract/service/TextExtractor.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.textract.TextractClient;
+import software.amazon.awssdk.services.textract.model.Block;
+import software.amazon.awssdk.services.textract.model.BlockType;
+import software.amazon.awssdk.services.textract.model.DetectDocumentTextResponse;
+
+@Service
+public class TextExtractor {
+
+    private TextractClient textractClient;
```
Comment (theangrydev) [inline aws-modules/amazon-textract/src/main/java/com/baeldung/textract/service/TextExtractor.java:18]:
```suggestion
    private final TextractClient textractClient;
```

File: aws-modules/amazon-textract/src/main/java/com/baeldung/textract/service/TextExtractor.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                    .bucket(bucketName)
+                    .name(objectKey)
+                    .build())
+                .build())
+            .build());
+        
+        return process(response);
+    }
+    
+    private String process(DetectDocumentTextResponse response) {
```
Comment (theangrydev) [inline aws-modules/amazon-textract/src/main/java/com/baeldung/textract/service/TextExtractor.java:48]: Let's rename this to be more descriptive of what it is doing

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: aws-modules/amazon-textract/src/main/java/com/baeldung/textract/service/TextExtractor.java (hunk 1 above)
Comment (hardikSinghBehl) [reply aws-modules/amazon-textract/src/main/java/com/baeldung/textract/service/TextExtractor.java]: review incorporated

File: aws-modules/amazon-textract/src/main/java/com/baeldung/textract/service/TextExtractor.java (hunk 2 above)
Comment (hardikSinghBehl) [reply aws-modules/amazon-textract/src/main/java/com/baeldung/textract/service/TextExtractor.java]: done, changed it to _transformTextDetectionResult()_.

**Review by theangrydev — APPROVED** [review]

**Review by theangrydev — APPROVED** [review]

**Pull request merged**